generators/csharp/codegen/src/asIs/RawClient.Template.cs

```diff
@@ -85,37 +122,50 @@ public record ApiResponse
     }
 
     private async Task<ApiResponse> SendWithRetriesAsync(
-        BaseApiRequest request,
+        HttpRequestMessage request,
+        IRequestOptions? options,
         CancellationToken cancellationToken
     )
     {
-        var httpClient = request.Options?.HttpClient ?? Options.HttpClient;
-        var maxRetries = request.Options?.MaxRetries ?? Options.MaxRetries;
-        var response = await httpClient
-            .SendAsync(BuildHttpRequest(request), cancellationToken)
-            .ConfigureAwait(false);
+        var httpClient = options?.HttpClient ?? Options.HttpClient;
+        var maxRetries = options?.MaxRetries ?? Options.MaxRetries;
+        var response = await httpClient.SendAsync(request, cancellationToken).ConfigureAwait(false);
         for (var i = 0; i < maxRetries; i++)
         {
             if (!ShouldRetry(response))
             {
                 break;
             }
+            AssertRetryableContent(request.Content);
+
             var delayMs = Math.Min(BaseRetryDelay * (int)Math.Pow(2, i), MaxRetryDelayMs);
             await SystemTask.Delay(delayMs, cancellationToken).ConfigureAwait(false);
+            using var retryRequest = CloneRequest(request);
             response = await httpClient
-                .SendAsync(BuildHttpRequest(request), cancellationToken)
+                .SendAsync(retryRequest, cancellationToken)
                 .ConfigureAwait(false);
         }
         return new ApiResponse { StatusCode = (int)response.StatusCode, Raw = response };
     }
 
+    private static void AssertRetryableContent(HttpContent requestContent)
+    {
+        switch (requestContent)
+        {
+            case StreamContent:
+                throw new Exception("Cannot retry a request with StreamContent");
+            case MultipartContent:
+                throw new Exception("Cannot retry a request with MultipartContent");
+        }
+    }
```

Comment:
Will the user see these errors? They don't control the default retry logic, so would it be better to simply not retry for such endpoints?

Reply:
Yes, fair!